probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -166,8 +166,15 @@ where
         // Offset of byte in word (little endian)
         let bit_offset = (address - aligned.start) * 8;
 
-        // Read 32-bit word and extract the correct byte
-        let result = ((self.read32(aligned.start)? >> bit_offset) & 0xFF) as u8;
+        let csw = self.build_csw_register(DataSize::U8);
```

Comment:
So, why we didn't implement true 8bit read/write initially is that not all chips support this.
So basically for this to work properly, we would need to poll the chip capabilities at some place first and check if 8bit is possible.

What is always possible is 32bit reads, which is why we used this.

Also, why do you still have the result with the shift? That would yield improper results I suppose since now you read the correct byte already and shift it's contenst like before. Am I mistaken?

Reply:
Shift is required because that's how data is written and read via the DRW register. And that's why tests do not pass :) MockMemoryAP implements this differently.